packages/eas-cli/src/project/android/gradleUtils.ts

```diff
@@ -0,0 +1,103 @@
+import { AndroidConfig } from '@expo/config-plugins';
+import fs from 'fs-extra';
+import g2js from 'gradle-to-js/lib/parser';
+
+// represents gradle command
+// e.g. for `:app:buildExampleDebug` -> { moduleName: app, flavor: example, buildType: debug }
+interface GradleCommand {
+  moduleName?: string;
+  flavor?: string;
+  buildType?: string;
+}
+
+interface Config {
+  applicationId?: string;
+  versionCode?: string;
+  versionName?: string;
+}
+
+interface AppBuildGradle {
+  android?: {
+    defaultConfig?: Config;
+    flavorDimensions?: string; // e.g. '"dimension1", "dimension2"'
+    productFlavors?: Record<string, Config>;
+  };
+}
+
+export const DEFAULT_MODULE_NAME = 'app';
+
+export async function getAppBuildGradleAsync(projectDir: string): Promise<AppBuildGradle> {
+  const buildGradlePath = AndroidConfig.Paths.getAppBuildGradleFilePath(projectDir);
+  const rawBuildGradle = await fs.readFile(buildGradlePath, 'utf8');
+  return await g2js.parseText(rawBuildGradle);
+}
+
+export function resolveConfigValue(
```

Comment:
How are you planning to support writing to multiflavor gradles?

e.g. what is the analog to this? https://github.com/expo/expo-cli/blob/master/packages/config-plugins/src/android/Version.ts#L40

In general is multiflavor support only intended for EAS projects. Why are these gradle utils are in `eas-cli` intead of `config-plugins`?

Reply:
>  Why are these gradle utils are in eas-cli intead of config-plugins?

https://github.com/expo/expo-cli/pull/3830/files#r704836238

> How are you planning to support writing to multiflavor gradles?

I don't think we should, we should just limit autoincrement functionality for managed and simple bare projects(by simple I mean no favors on android and one scheme/target on iOS). With projects like that, we are likely to cause more problems than the feature is worth.

Reply:
OK cool, seems like we are on the same page with both issues. 

Will continue monitoring discussion on your slack post.